Mathlib/SetTheory/Game/PGame.lean

```diff
@@ -150,6 +151,15 @@ theorem rightMoves_mk {xl xr xL xR} : (⟨xl, xr, xL, xR⟩ : PGame).RightMoves
 theorem moveRight_mk {xl xr xL xR} : (⟨xl, xr, xL, xR⟩ : PGame).moveRight = xR :=
   rfl
 
+lemma ext {x y : PGame} (hl : x.LeftMoves = y.LeftMoves) (hr : x.RightMoves = y.RightMoves)
+    (hL : ∀ i j, HEq i j → x.moveLeft i = y.moveLeft j)
+    (hR : ∀ i j, HEq i j → x.moveRight i = y.moveRight j) :
+    x = y := by
+  suffices HEq x.moveLeft y.moveLeft → HEq x.moveRight y.moveRight → x = y from
+    this (hfunext hl (heq_of_eq <| hL · · ·)) (hfunext hr (heq_of_eq <| hR · · ·))
+  cases x; cases y; cases hl; cases hr
+  rintro ⟨rfl⟩ ⟨rfl⟩; rfl
```

Comment:
```suggestion
  cases x
  cases y
  subst hl hr
  simp only [leftMoves_mk, rightMoves_mk, heq_eq_eq, forall_eq', mk.injEq, true_and] at *
  exact ⟨funext hL, funext hR⟩
```
Here's an alternate proof which avoids the awkward auxiliary `suffices` block. Again, feel free to make style tweaks if you want.